### PR TITLE
Remove or rework assumptions in queue proofs

### PR DESCRIPTION
**Pull request by dan4thewin**
This commit is paired with another to queue.c in the kernel.  To accommodate significant performance changes in newer versions of CBMC, the --pointer-overflow-check flag is removed.  The --pointer-check and --bounds-check flags remain to verify pointers upon dereference.  The configASSERT() macro is changed to evaluate to nothing allowing CBMC to more fully check the code, rather than the prior practice of adding assertions to evade asserts.

These proofs all pass without errors, and run in a small amount of time.

### CODE REVIEW

File: FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreate/QueueGenericCreate_harness.c [hunk 1]
```diff
@@ -32,19 +32,14 @@
...
+void harness()
+{
+    UBaseType_t uxQueueLength;
+    UBaseType_t uxItemSize;
+    uint8_t ucQueueType;
 
-	size_t uxQueueStorageSize;
-	__CPROVER_assume(uxQueueStorageSize < (UINT32_MAX>>8));
+    /* Allow CBMC to run in a reasonable amount of time. */
+    __CPROVER_assume( ( uxQueueLength == QUEUE_LENGTH ) || ( uxItemSize == QUEUE_ITEM_SIZE ) );
```
Comment (AniruddhaKanhere) [inline FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreate/QueueGenericCreate_harness.c:42]: This assumption seems very restrictive...

If there is no other way, can we at least define the `QUEUE_LENGTH` and `QUEUE_ITEM_SIZE` to various values in the configurations file above? That would broaden the scope of the proofs.

File: FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreateStatic/QueueGenericCreateStatic_harness.c [hunk 2]
```diff
@@ -31,32 +31,22 @@
...
-
     uint8_t ucQueueType;
+    size_t storageSize;
 
-    __CPROVER_assume(uxQueueStorageSize < (UINT32_MAX>>8));
-
-    // QueueGenericReset does not check for multiplication overflow
-    __CPROVER_assume(uxItemSize < uxQueueStorageSize/uxQueueLength);
+    /* Allow CBMC to run in a reasonable amount of time. */
+    __CPROVER_assume( ( uxQueueLength == QUEUE_LENGTH ) || ( uxItemSize == QUEUE_ITEM_SIZE ) );
```
Comment (AniruddhaKanhere) [inline FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreateStatic/QueueGenericCreateStatic_harness.c:42]: Similar comment as above.

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreate/QueueGenericCreate_harness.c (hunk 1 above)
Comment (dan4thewin) [reply FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreate/QueueGenericCreate_harness.c:42]: It's binding one or the other, but not both at once; i.e., when queue length is fixed, item size can be anything, and when item size is fixed, queue length can be anything.  I used macros with the intent of providing additional values in a future PR.

Comment (AniruddhaKanhere) [reply FreeRTOS/Test/CBMC/proofs/Queue/QueueGenericCreate/QueueGenericCreate_harness.c:42]: Makes sense. Thanks.

**Review by markrtuttle — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Comment by dan4thewin** [comment]
Thanks to Kareem and Oscar, we have evidence that the proofs CI checks are passing.  I'll merge this PR now.

**Pull request merged**